test organization

haxscramper opened the pull request:
Follow-up for https://github.com/nim-works/nimskull/pull/461, mostly for imports and some docs

- Replace `import tables`  and `import sets` in tests  (replacement is done using `fd | sd`, so it is not exhaustive yet).
- Handle `testament  cat stdlib/os` properly --  there is no reason  to ban this as  (surprise!) test  directory might  have more  than one  level. I guess level of  orderliness of the core maintainers shows  in that pretty well.  After fifteen  years nobody  had ever  thought that  tests can  be placed into something other than 133+ entries on the same level.
- Add  small entry  for  using  `koch.py` for  test  running  tests in  the documentation.
- Add missing testament link from the documentation
- Comment out `nimpretty` link as it was removed earlier.

saem commented:
I think there is some special handling within `categories` inside `processCategory` proc at the very least, that and others need to become aware of categories now having subdirs. Otherwise it's on the right track.

haxscramper committed:
Test documentation, handle slash in category name

- Handle `testament  cat stdlib/os` properly --  there is no reason  to ban
  this as  (surprise!) test  directory might  have more  than one  level. I
  guess level of  orderliness of the core maintainers shows  in that pretty
  well.  After fifteen  years nobody  had ever  thought that  tests can  be
  placed into something other than 133+ entries on the same level.
- Add  small entry  for  using  `koch.py` for  test  running  tests in  the
  documentation.
- Add missing testament link from the documentation
- Comment out `nimpretty` link as it was removed earlier.

haxscramper commented:
bors r+
